### ISSUE TITLE
Runtime Inference API delegates model execution to caller-provided `next_logits` instead of owning graph → dispatch → Provider execution

### ISSUE THREAD

**Issue by astorise** [astorise/Magnetar#26]
## Priority

**P0 — blocks `v0.1`**

## Context

The Runtime Inference API is intended to own inference execution.

According to the accepted OpenSpec architecture, the expected execution path is:

```text
Runtime Inference API
  -> Model Instance
  -> Model Component
  -> Execution Graph
  -> Operator validation
  -> Kernel Registry
  -> Kernel Dispatch
  -> Provider
  -> TensorResource logits
  -> Sampling
```

The current implementation does not yet execute this path end-to-end.

## Problem

`run_generation_loop()` currently receives execution state and logits computation from its caller:

```rust
provider_ready: bool,
kernel_ready: bool,
next_logits: impl FnMut(&[TokenId]) -> Vec<f32>,
```

This means the caller:

- tells Runtime whether a Provider is ready;
- tells Runtime whether a Kernel is ready;
- computes the logits outside Runtime;
- effectively owns the Model/Provider execution boundary.

Runtime then only owns generation orchestration and sampling.

This conflicts with the Runtime Inference API contract: Provider/Device resolution and model execution are Runtime responsibilities.

It also makes it possible for callers to bypass:

- Model Component execution;
- Execution Graph validation;
- Kernel Registry;
- Kernel Dispatch;
- Resource Affinity;
- Provider readiness revalidation;
- Memory Manager-controlled TensorResources.

## Current effective path

```text
caller
  ├── provider_ready = true
  ├── kernel_ready = true
  └── next_logits()
          |
          v
      Generation
          |
          v
       Sampling
```

## Expected path

```text
caller
   |
   v
Runtime Inference API
   |
   v
Runtime execution orchestration
   |
   +-> Model Instance
   +-> Model Component
   +-> Execution Graph
   +-> Kernel Registry
   +-> Kernel Dispatch
   +-> Provider
   +-> Memory Manager / TensorResource
   |
   v
logits
   |
   v
Sampling
```

## Required change

Introduce a Runtime-owned execution path that can produce logits from a loaded Model Instance.

The high-level inference API should no longer require callers to provide:

```rust
provider_ready
kernel_ready
next_logits
```

Instead Runtime should resolve and execute the required graph itself.

A possible internal shape is:

```rust
Runtime::execute_generation_step(...)
    -> ModelInstance
    -> ModelComponent graph
    -> validate graph
    -> resolve operators
    -> KernelRegistry
    -> KernelDispatch
    -> Provider
    -> TensorResource
    -> logits
```

Exact internal APIs are implementation-defined.

## Requirements

- Runtime owns Provider readiness evaluation.
- Runtime owns Device compatibility evaluation.
- Runtime owns Kernel Registry selection.
- Runtime owns dispatch.
- Runtime owns Resource Affinity checks.
- Runtime owns Memory Manager feasibility checks.
- Runtime owns TensorResource lifecycle.
- Model Component does not directly select Provider/Device/Kernel.
- Caller cannot supply executable logits callbacks to normal inference.
- No raw Provider/Device/Kernel handles are exposed.
- No raw tensor pointers are exposed.
- No silent CPU fallback is introduced.

## CLI impact

`magnetar-cli` currently uses placeholder logits to exercise the Runtime API boundary.

Once this issue is complete:

```text
magnetar run
    -> Runtime Inference API
    -> real Runtime execution path
```

should replace placeholder logits for the baseline fixture.

## Acceptance criteria

- [ ] Normal generation no longer requires caller-supplied `provider_ready`.
- [ ] Normal generation no longer requires caller-supplied `kernel_ready`.
- [ ] Normal generation no longer requires caller-supplied `next_logits`.
- [ ] Runtime selects an eligible Kernel through Kernel Registry.
- [ ] Kernel Dispatch invokes Reference CPU Provider for the baseline.
- [ ] Resulting logits are represented through Runtime-owned tensor/resource contracts.
- [ ] Sampling consumes logits produced by the Runtime execution path.
- [ ] Provider readiness is checked at selection/dispatch time.
- [ ] Missing compatible Kernel returns a structured error.
- [ ] Missing/failed Provider returns a structured error.
- [ ] Memory admission failure returns a structured error.
- [ ] Existing Runtime authority boundaries remain unchanged.
- [ ] Contract tests cover success and failure paths.

## OpenSpec coverage

This is an **implementation gap**, not a new architecture decision.

Relevant accepted contracts include:

- `define-runtime-inference-api`
- `define-execution-graph-and-operator-contract`
- `define-kernel-contract`
- `define-kernel-registry-and-dispatch`
- `define-reference-cpu-provider-and-kernel-baseline`
- `define-end-to-end-local-inference-conformance`

No new OpenSpec change should be required unless implementation reveals a missing semantic decision.

## Blocks

`v0.1`

**Comment by astorise** [comment]
Fixed in PR #33 via commit e5be180. Normal generation no longer accepts caller-supplied provider_ready, kernel_ready, or next_logits inputs; logits now come from a Runtime-registered generation executor with structured execution evidence for graph validation, kernel selection/dispatch, provider execution, and tensor logits.
